models/repo/repo.go

```diff
@@ -654,6 +664,14 @@ func (repo *Repository) GetTrustModel() TrustModelType {
 	return trustModel
 }
 
+// GetIsArchivedError returns ErrRepoIsArchived is the repo is archived
+func (repo *Repository) GetIsArchivedError() error {
+	if repo.IsArchived {
+		return ErrRepoIsArchived{ID: repo.ID, Name: repo.Name, Owner: repo.OwnerName}
```

Comment:
I recommend against doing that, instead pass the repo directly, and then use `repo.LogString()` inside the error message.